src/pycram/orm/utils.py

```diff
@@ -37,44 +38,86 @@ def print_database(in_Sessionmaker: sqlalchemy.orm.sessionmaker):
         rospy.loginfo(result)
 
 
-def update_primary_key(source_engine: sqlalchemy.orm.engine, destination_engine: sqlalchemy.orm.engine):
+def get_all_children_set(in_node, node_set=None):
+    all_nodes = set()
+    if node_set is None:
+        node_set = set()
+    node_set.add(in_node)
+    if in_node.__subclasses__():
+        for node in in_node.__subclasses__():
+            all_nodes.update(get_all_children_set(node, node_set))
+    else:
+        pass
+        all_nodes.update(node_set)
+    return all_nodes
+
+
+def get_tree(in_node,parent=None,tree=None):
+    if parent is None:
+        node=Node(in_node)
+        tree=node
+    else:
+        node=Node(in_node,parent)
+    if len(in_node.__subclasses__()):
+        for subnode in in_node.__subclasses__():
+            get_tree(subnode,node,tree)
+    return tree
+
+
+def update_primary_key(source_session_maker: sqlalchemy.orm.sessionmaker,
```

Comment:
functions should do one thing. This function can be partitioned in a lot more functions